core/modules/field_ui/field_ui.module

```diff
@@ -119,39 +131,77 @@ function field_ui_menu() {
           $weight = 0;
           $view_modes = array('default' => array('label' => t('Default'))) + $entity_info['view modes'];
           foreach ($view_modes as $view_mode => $view_mode_info) {
             $items["$path/display/$view_mode"] = array(
               'title' => $view_mode_info['label'],
-              'page arguments' => array('field_ui_display_overview_form', $entity_type, $bundle_arg, $view_mode),
+              'page callback' => 'backdrop_get_form',
+              'page arguments' => array('field_ui_display_form', $entity_type, $bundle_arg, $view_mode),
               // The access callback needs to check both the current 'custom
               // display' setting for the view mode, and the overall access
               // rules for the bundle admin pages.
               'access callback' => '_field_ui_view_mode_menu_access',
               'access arguments' => array_merge(array($entity_type, $bundle_arg, $view_mode, $access['access callback']), $access['access arguments']),
-              'type' => ($view_mode == 'default' ? MENU_DEFAULT_LOCAL_TASK : MENU_LOCAL_TASK),
               'weight' => ($view_mode == 'default' ? -10 : $weight++),
               'file' => 'field_ui.admin.inc',
             );
           }
+
+          // View mode administration.
+          $items["$path/display/view-modes/add"] = array(
+            'title' => 'Add view mode',
+            'page callback' => 'backdrop_get_form',
+            'page arguments' => array('field_ui_view_mode_form', $entity_type, $bundle_arg),
+            'access arguments' => array('administer view modes'),
+            'file' => 'field_ui.admin.inc',
+            'type' => MENU_LOCAL_ACTION,
+          );
+          $items["$path/display/view-modes/%/configure"] = array(
+            'title' => 'Configure view mode',
+            'page callback' => 'backdrop_get_form',
+            'page arguments' => array('field_ui_view_mode_form', $entity_type, $bundle_arg, $view_mode_position),
+            'access arguments' => array('administer view modes'),
+            'file' => 'field_ui.admin.inc',
+          );
+          $items["$path/display/view-modes/%/delete"] = array(
+            'title' => 'Delete view mode',
+            'page callback' => 'backdrop_get_form',
+            'page arguments' => array('field_ui_view_mode_delete_form', $entity_type, $bundle_arg, $view_mode_position),
+            'access arguments' => array('administer view modes'),
+            'file' => 'field_ui.admin.inc',
+          );
+          $items["$path/display/view-modes/%/enable"] = array(
+            'title' => 'Enable view mode',
+            'page callback' => 'field_ui_view_mode_enable',
+            'page arguments' => array($entity_type, $bundle_arg, $view_mode_position),
+            'access arguments' => array('administer view modes'),
+            'file' => 'field_ui.admin.inc',
+          );
+          $items["$path/display/view-modes/%/reset"] = array(
```

Comment:
Some view modes are provided by modules, and others are provided in config. The loader we have now only loads them from config, so that approach would fail for things like `full` and `teaser`.